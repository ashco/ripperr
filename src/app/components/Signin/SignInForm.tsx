import React, { useState, useContext } from 'react';
import { useRouter } from 'next/router';
import styled from 'styled-components';
import {
  Formik,
  Form,
  Field,
  ErrorMessage,
  useField,
  FormikProps,
} from 'formik';
import * as Yup from 'yup';

import { FirebaseContext } from '../Firebase';
import { IError } from '../Signup/SignUpForm';

interface IFormValues {
  email: string;
  password: string;
  // error: null | IError;
}

const INITIAL_FORM_VALUES: IFormValues = {
  email: '',
  password: '',
  // error: null,
};

const SignInForm = () => {
  const firebase = useContext(FirebaseContext);
  const router = useRouter();

  return (
    <Formik
      initialValues={INITIAL_FORM_VALUES}
      validationSchema={Yup.object({
        email: Yup.string()
          .email('Invalid email address')
          .required('Required'),
      })}
      onSubmit={({ email, password }, { resetForm }) => {
        firebase
          .doSignInWithEmailAndPassword(email, password)
          .then(() => {
            resetForm();
            router.push('/');
          })
          .catch(error => {
            console.log(error);
          });
      }}
    >
      <Form>
        {/* <TextInput
          label="Email"
          name="email"
          type="email"
          placeholder="janedoe@gmail.com"
        /> */}
        <label htmlFor="email">Password</label>
        <Field name="email" type="email" />
        <ErrorMessage name="email" />
        <label htmlFor="password">Password</label>
        <Field name="password" type="password" />
        <ErrorMessage name="password" />
        <button type="submit">Submit</button>
      </Form>
    </Formik>
  );
};

// // Shape of form values
// interface FormValues {
//   email: string;
//   password: string;
// }

// interface OtherProps {
//   message: string;
// }

// const InnerForm = (props: OtherProps & FormikProps<FormValues>) => {
//   const { touched, errors, isSubmitting, message } = props;
//   return (
//     <Form>
//       <h1>{message}</h1>
//       <Field type="email" name="email" />
//       {touched.email && errors.email && <div>{errors.email}</div>}

//       <Field type="password" name="password" />
//       {touched.password && errors.password && <div>{errors.password}</div>}

//       <button type="submit" disabled={isSubmitting}>
//         Submit
//       </button>
//     </Form>
//   );
// };

// interface FormValues = {

// }

// const TextInput = (props: FormikProps<FormValues>) => {
//   // useField() returns [formik.getFieldProps(), formik.getFieldMeta()]
//   // which we can spread on <input> and also replace ErrorMessage entirely.
//   const [field, meta] = useField(props);
//   return (
//     <>
//       <label htmlFor={props.id || props.name}>{label}</label>
//       <input {...field} {...props} />
//       {meta.touched && meta.error ? (
//         <div className="error">{meta.error}</div>
//       ) : null}
//     </>
//   );
// };

const FormWrapper = styled.form`
  /* border: black 4px solid; */
  display: flex;
  flex-direction: column;
  * {
    margin-bottom: 4px;
  }
  input {
    height: 2rem;
  }
`;

export default SignInForm;
