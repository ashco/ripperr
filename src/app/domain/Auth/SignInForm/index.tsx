import React, { useState, useContext } from 'react';
import { useRouter } from 'next/router';
import styled from 'styled-components';
import { Formik, Form } from 'formik';

import InputField from '@/domain/MovementForm/InputField';
import { signInVal } from '@/domain/MovementForm/validationSchema';
import FormError from '@/components/FormError';
import Button from '@/components/Button';

import { FirebaseContext } from '@/context';
import { IAuthError } from '@/types/types';

export interface ISignInFormValues {
  email: string;
  password: string;
}

const INITIAL_VALUES: ISignInFormValues = {
  email: '',
  password: '',
};

const SignInForm = () => {
  const firebase = useContext(FirebaseContext);
  const router = useRouter();

  const [error, setError] = useState<IAuthError | false>(false);

  return (
    <Formik
      initialValues={INITIAL_VALUES}
      validationSchema={signInVal}
      onSubmit={({ email, password }, { resetForm }) => {
        firebase
          .doSignInWithEmailAndPassword(email, password)
          .then(() => {
            resetForm();
            router.push('/');
          })
          .catch((error) => {
            setError(error);
            console.log(error);
          });
      }}
    >
      <StyledForm>
        <InputField
          // label="Email"
          name="email"
          type="email"
          placeholder="Email"
        />
        <InputField
          // label="Password"
          name="password"
          type="password"
          placeholder="Password"
        />
        <Button type="submit">Sign In</Button>
        {error && <FormError>{error.message}</FormError>}
      </StyledForm>
    </Formik>
  );
};

const StyledForm = styled(Form)`
  /* border: black 4px solid; */
  display: grid;
  grid-template-rows: auto auto auto;
  /* * {
    margin-bottom: 4px;
  } */
  /* input {
    height: 2rem;
  } */
`;

export default SignInForm;
