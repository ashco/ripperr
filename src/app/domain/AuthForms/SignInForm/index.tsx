import React, { useState, useContext } from 'react';
import { useRouter } from 'next/router';
import styled from 'styled-components';
import { Formik, Form } from 'formik';

import InputField from './node_modules/components/MovementForm/InputField';
import { signInVal } from './node_modules/components/MovementForm/validationSchema';
import FormError from './node_modules/components/FormError';
import Button from './node_modules/components/Button';

import { FirebaseContext } from './node_modules/context';
import { IAuthError } from './node_modules/types/types';

export interface ISignInFormValues {
  email: string;
  password: string;
}

const INITIAL_VALUES: ISignInFormValues = {
  email: '',
  password: '',
};

const SignInForm = () => {
  const firebase = useContext(FirebaseContext);
  const router = useRouter();

  const [error, setError] = useState<IAuthError | false>(false);

  return (
    <Formik
      initialValues={INITIAL_VALUES}
      validationSchema={signInVal}
      onSubmit={({ email, password }, { resetForm }) => {
        firebase
          .doSignInWithEmailAndPassword(email, password)
          .then(() => {
            resetForm();
            router.push('/');
          })
          .catch((error) => {
            setError(error);
            console.log(error);
          });
      }}
    >
      <StyledForm>
        <InputField
          // label="Email"
          name="email"
          type="email"
          placeholder="Email"
        />
        <InputField
          // label="Password"
          name="password"
          type="password"
          placeholder="Password"
        />
        <Button type="submit">Sign In</Button>
        {error && <FormError>{error.message}</FormError>}
      </StyledForm>
    </Formik>
  );
};

const StyledForm = styled(Form)`
  /* border: black 4px solid; */
  display: grid;
  grid-template-rows: auto auto auto;
  /* * {
    margin-bottom: 4px;
  } */
  /* input {
    height: 2rem;
  } */
`;

export default SignInForm;
