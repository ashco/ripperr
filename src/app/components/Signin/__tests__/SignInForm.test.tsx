import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import SignInForm from '../SignInForm';

describe('SignInForm', () => {
  // it('My Test Case', () => {
  //   expect(true).toEqual(true);
  // });

  it('snapshot renders', () => {
    const component = renderer.create(<SignInForm />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows sign in button', () => {
    const wrapper = mount(<SignInForm />);
    expect(wrapper.find('button').text()).toEqual('Sign In');
  });
});
