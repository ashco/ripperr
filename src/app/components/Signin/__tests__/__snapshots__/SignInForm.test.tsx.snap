// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInForm snapshot renders 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <label
    htmlFor="email"
  >
    <div>
      Email
    </div>
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="janedoe@gmail.com"
      type="email"
      value=""
    />
  </label>
  <label
    htmlFor="password"
  >
    <div>
      Password
    </div>
    <input
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="password123"
      type="password"
      value=""
    />
  </label>
  <button
    type="submit"
  >
    Sign In
  </button>
</form>
`;
